skrifa/src/color/instance.rs

```diff
@@ -0,0 +1,660 @@
+    /// applied.
+    pub fn resolve(
+        &self,
+        instance: &'a ColrInstance<'a>,
+    ) -> impl Iterator<Item = ResolvedColorStop> + 'a {
+        self.stops
+            .iter()
+            .map(|stop| ResolvedColorStop {
+                offset: stop.stop_offset().to_f32(),
+                palette_index: stop.palette_index(),
+                alpha: stop.alpha().to_f32(),
+            })
+            .chain(self.var_stops.iter().map(|stop| {
+                let deltas = instance.var_deltas(stop.var_index_base(), [DeltaType::F2Dot14; 2]);
+                ResolvedColorStop {
+                    offset: stop.stop_offset().apply_delta(deltas[0]),
+                    palette_index: stop.palette_index(),
+                    alpha: stop.alpha().apply_delta(deltas[1]),
+                }
+            }))
+    }
+}
+
+/// Simplified version of `Paint` with applied variation deltas.
+///
+/// These are constructed with the [`resolve_paint`] function.
+///
+/// This is roughly equivalent to FreeType's
+/// [`FT_COLR_Paint`](https://freetype.org/freetype2/docs/reference/ft2-layer_management.html#ft_colr_paint)
+/// type.
+pub enum ResolvedPaint<'a> {
+    ColrLayers {
+        range: Range<usize>,
+    },
+    Solid {
+        palette_index: u16,
+        alpha: f32,
+    },
+    LinearGradient {
+        x0: f32,
+        y0: f32,
+        x1: f32,
+        y1: f32,
+        x2: f32,
+        y2: f32,
+        color_stops: ColorStops<'a>,
+        extend: Extend,
+    },
+    RadialGradient {
+        x0: f32,
+        y0: f32,
+        radius0: f32,
+        x1: f32,
+        y1: f32,
+        radius1: f32,
+        color_stops: ColorStops<'a>,
+        extend: Extend,
+    },
+    SweepGradient {
+        center_x: f32,
+        center_y: f32,
+        start_angle: f32,
+        end_angle: f32,
+        color_stops: ColorStops<'a>,
+        extend: Extend,
+    },
+    Glyph {
+        glyph_id: GlyphId,
+        paint: Paint<'a>,
+    },
+    ColrGlyph {
+        glyph_id: GlyphId,
+    },
+    Transform {
+        xx: f32,
+        yx: f32,
+        xy: f32,
+        yy: f32,
+        dx: f32,
+        dy: f32,
+        paint: Paint<'a>,
+    },
+    Translate {
+        dx: f32,
+        dy: f32,
+        paint: Paint<'a>,
+    },
+    Scale {
+        scale_x: f32,
+        scale_y: f32,
+        around_center: Option<Point<f32>>,
+        paint: Paint<'a>,
+    },
+    Rotate {
+        angle: f32,
+        around_center: Option<Point<f32>>,
+        paint: Paint<'a>,
+    },
+    Skew {
+        x_skew_angle: f32,
+        y_skew_angle: f32,
+        around_center: Option<Point<f32>>,
+        paint: Paint<'a>,
+    },
+    Composite {
+        source_paint: Paint<'a>,
+        mode: CompositeMode,
+        backdrop_paint: Paint<'a>,
+    },
+}
+
+/// Resolves this paint with the given instance.
+///
+/// Resolving means that all numeric values are converted to 16.16 fixed
```

Comment:
This comment could perhaps be rephrased or updated in terms of how and when the conversion to float happens?

Reply:
Oops, good catch.

Reply:
Updated the comment to reflect the floating point values.